learn/work/fairness-detectors/index.qmd

```diff
@@ -0,0 +1,290 @@
+:::
+
+<!-- TODO: remove the /dev from the above link once this yardstick vignette makes it to CRAN -->
+
+From this perspective, the models with the highest `roc_auc()` estimates are the most fair.
+
+The fairness assessment for this stakeholder doesn't need any of the functionality newly introduced in yardstick 1.3.0. An analysis that reconciles the role of the `native` variable in these models' predictions will, though.
+
+## Fair prediction on human-written essays
+
+Now, consider that our only priority was to correctly classify human-written text as human-written and incorrectly classify human-written text as generated by AI at the same rate for both native English writers and non-native English writers, ignoring predictions from essays generated by AI.
+
+-   Another **student** whose work is evaluated by a detector model may take on such a perspective. This student could be a native English writer who does not want other students to be subjected to undue harm or a non-native English writer concerned with their own writing being incorrectly classified as GPT-generated.
+
+-   Another **instructor** of a course may feel it is unfair to disproportionately classify writing from students that are non-native English writers as GPT-generated, regardless of how effectively the model detects GPT-generated essays.
+
+The `equal_opportunity()` metric enables us to quantify the extent of this interpretation of unfairness. Equal opportunity is satisfied when a model's predictions have the same true positive and false negative rates across protected groups; a model predicts more fairly if it's equally likely to predict a positive outcome for each group.
+
+In this example, a GPT detector satisfies equal opportunity when the detector correctly classifies human-written text as human-written and incorrectly classifies human-written text as generated by AI at the same rate for both native English writers and non-native English writers.
+
+::: callout-note
+This definition does not consider the predictions based on essays generated by AI.
+:::
+
+To calculate equal opportunity with yardstick, we use the `equal_opportunity()` function:
+
+```{r ep-opp-native}
+equal_opportunity_by_native <- equal_opportunity(by = native)
+```
+
+<!-- TODO: print out the following function once the metric print method is merged into yardstick -->
+
+The function `equal_opportunity_by_native()` is a yardstick metric function like any other, except it knows to temporarily group by and summarize across a data-column called `native`. Applying it:
+
+```{r eq-opp}
+detectors %>%
+  filter(kind == "Human") %>%
+  group_by(detector) %>%
+  equal_opportunity_by_native(truth = kind, estimate = .pred_class, event_level = "second") %>%
+  arrange(.estimate) %>%
+  head(3)
+```
+
+The detectors with estimates closest to zero are most fair, by this definition of fairness.
+
+::: callout-important
+Why do we need to specify `event_level = "second"`?
```

Comment:
I suggest you add a link here so folks can look this up. What's the best link that explains this in the yardstick documentation?

Reply:
This is a good idea. It doesn't appear to me that there are any dedicated help-files in yardstick docs about this, so maybe we link to the "Relevant level" section of the yardstick function behind the scenes in `equal_opportunity()`? At https://yardstick.tidymodels.org/dev/reference/sens.html#relevant-level

Reply:
yes, that appears to be the best place to link to at the moment